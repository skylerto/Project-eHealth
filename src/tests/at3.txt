-- at2.txt: Ordering of prescriptions

-- Add a physician
add_physician	   (1, "Mayo", specialist)

-- Add a patient
add_patient        (3, "Drew")
add_patient        (1, "Helen")


--add medications, not in order off ID
add_medication     (1, ["Wafarin", pill, 1.0, 6.0])
add_medication     (3, ["caffeine", liquid, 7.0, 16.0])
add_medication     (2, ["acetaminophen", liquid, 10.0, 25.5])
-- How are medications displayed?

add_interaction    (1,2)
add_interaction    (1,3)
add_interaction    (2,3)
--interactions are displayed alphabetically by name

-- prescription id 1: generalist Mayo prescribes for DORA
new_prescription   (2, 1, 3)
-- prescription id 1: generalist Mayo prescribes for DREW
new_prescription   (1, 1, 1)
-- TEST ORDER OF PRESCRIPTIONS

dpr_q

-- TEST BY ALPHA

-- ADD FOR DREW
-- add Wafarin 5.5 mg to prescription id 1 "DREW"
add_medicine       (1, 1, 5.5)
-- add caffeine 300.5 mg to prescription id 1 "DREW"
add_medicine       (1, 2, 300.5)
-- add acetaminophen 5.5 mg to prescription id 1 "DREW"
add_medicine       (1, 3, 5.5)
-- ADD MEDICINE IN ID ORDER

dpr_q

-- ADD FOR DORA
-- add Wafarin 5.5 mg to prescription id 1 "DREW"
add_medicine       (2, 2, 5.5)
-- add caffeine 300.5 mg to prescription id 1 "DREW"
add_medicine       (2, 3, 300.5)
-- add acetaminophen 5.5 mg to prescription id 1 "DREW"
add_medicine       (2, 1, 5.5)
-- ADD MEDICINE IN ID ALPHA

dpr_q

remove_medicine(1,1)
remove_medicine(1,2)
remove_medicine(1,3)

remove_medicine(2,1)
remove_medicine(2,2)
remove_medicine(2,3)

dpr_q

-- REVERSE ADDITIONS, TEST BY ID

-- ADD FOR DORA
-- add Wafarin 5.5 mg to prescription id 1 "DREW"
add_medicine       (2, 2, 5.5)
-- add caffeine 300.5 mg to prescription id 1 "DREW"
add_medicine       (2, 3, 300.5)
-- add acetaminophen 5.5 mg to prescription id 1 "DREW"
add_medicine       (2, 1, 5.5)
-- ADD MEDICINE IN ID ALPHA

dpr_q

-- ADD FOR DREW
-- add Wafarin 5.5 mg to prescription id 1 "DREW"
add_medicine       (1, 1, 5.5)
-- add caffeine 300.5 mg to prescription id 1 "DREW"
add_medicine       (1, 2, 300.5)
-- add acetaminophen 5.5 mg to prescription id 1 "DREW"
add_medicine       (1, 3, 5.5)
-- ADD MEDICINE IN ID ORDER

dpr_q
