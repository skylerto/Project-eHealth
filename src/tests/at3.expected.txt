  0: ok
  Physicians:
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_physician(1,"Mayo",specialist)
  1: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(3,"Drew")
  2: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    3->Drew
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(1,"Helen")
  3: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
  Interactions: 
  Prescriptions:
->add_medication(1,["Wafarin", pill, 1, 6])
  4: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
  Interactions: 
  Prescriptions:
->add_medication(3,["caffeine", liquid, 7, 16])
  5: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
  Prescriptions:
->add_medication(2,["acetaminophen", liquid, 10, 25.5])
  6: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
  Prescriptions:
->add_interaction(1,2)
  7: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
  Prescriptions:
->add_interaction(1,3)
  8: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
  Prescriptions:
->add_interaction(2,3)
  9: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
->new_prescription(2,1,3)
  10: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    2->[2,1,3,()]
->new_prescription(1,1,1)
  11: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,()]
    2->[2,1,3,()]
->dpr_q
  12: ok
  There are no dangerous prescriptions
->add_medicine(1,1,5.5)
  13: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50)]
    2->[2,1,3,()]
->add_medicine(1,2,300.5)
  14: dose is outside allowed range
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50)]
    2->[2,1,3,()]
->add_medicine(1,3,5.5)
  15: dose is outside allowed range
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50)]
    2->[2,1,3,()]
->dpr_q
  16: ok
  There are no dangerous prescriptions
->add_medicine(2,2,5.5)
  17: dose is outside allowed range
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50)]
    2->[2,1,3,()]
->add_medicine(2,3,300.5)
  18: dose is outside allowed range
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50)]
    2->[2,1,3,()]
->add_medicine(2,1,5.5)
  19: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50)]
    2->[2,1,3,(1->5.50)]
->dpr_q
  20: ok
  There are no dangerous prescriptions
->remove_medicine(1,1)
  21: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,()]
    2->[2,1,3,(1->5.50)]
->remove_medicine(1,2)
  22: medication is not in the prescription
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,()]
    2->[2,1,3,(1->5.50)]
->remove_medicine(1,3)
  23: medication is not in the prescription
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,()]
    2->[2,1,3,(1->5.50)]
->remove_medicine(2,1)
  24: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,()]
    2->[2,1,3,()]
->remove_medicine(2,2)
  25: medication is not in the prescription
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,()]
    2->[2,1,3,()]
->remove_medicine(2,3)
  26: medication is not in the prescription
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,()]
    2->[2,1,3,()]
->dpr_q
  27: ok
  There are no dangerous prescriptions
->add_medicine(2,2,5.5)
  28: dose is outside allowed range
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,()]
    2->[2,1,3,()]
->add_medicine(2,3,300.5)
  29: dose is outside allowed range
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,()]
    2->[2,1,3,()]
->add_medicine(2,1,5.5)
  30: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,()]
    2->[2,1,3,(1->5.50)]
->dpr_q
  31: ok
  There are no dangerous prescriptions
->add_medicine(1,1,5.5)
  32: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50)]
    2->[2,1,3,(1->5.50)]
->add_medicine(1,2,300.5)
  33: dose is outside allowed range
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50)]
    2->[2,1,3,(1->5.50)]
->add_medicine(1,3,5.5)
  34: dose is outside allowed range
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,10.00,25.50]
    3->[caffeine,lq,7.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50)]
    2->[2,1,3,(1->5.50)]
->dpr_q
  35: ok
  There are no dangerous prescriptions
